docs/docs/guides/04-fe-coding-guidelines.mdx

```diff
@@ -0,0 +1,596 @@
+Firefox ESR
+last 2 Opera versions
+not dead
+```
+
+It's important to note that this informs the JS transpilation step by esbuild, so if a browser does
+not support it, it _may_ be polyfilled.
+
+### A special note on `lodash`
+
+We target modern browsers, typically speaking these browsers have all the functionality lodash
+provides natively, albeit there may be some discrepancies between the functionality or API
+signature. Because of this and our general view of shipping as little javascript to the client as
+possible we should **never be adding a `lodash` function to our code**. In addition, when working on
+the codebase everyone should strive to remove the instances we currently have.
+
+A common use case is the `get` function:
+
+```jsx
+import get from 'lodash/get';
+
+const extractDeepAttribute = (schema: object, path: string[]) => {
+  return get(schema, ['attributes', ...path], {});
+};
+```
+
+But if we zoom out we can begin to question a few things:
+
+- why are we extracting something so deep here?
+- are my data structures the issue? i.e. to complex
+- can i just use
+  [`optionalChaining`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)?
+
+Typically, I think you'll find that rethinking how the code works will reveal an easier to manage
+data structure that from there you can probably use `optionalChaining`.
+
+If you're struggling to convert a `lodash` usage to native code, check out
+[this helpful guide](https://github.com/you-dont-need/You-Dont-Need-Lodash-Underscore).
+
+### Adding new dependencies
+
+Sometimes you require a new dependency, it happens, it's common. Third party library code is
+_fantastic_ because typically it's somewhat battle tested in production environments and in some
+circumstances where the concept is complicated e.g. `fractional-indexing` it really is easier to use
+a library than re-write and maintain in our application. It's a case by case basis therefore, when
+adding a new dependency to the codebase you should do the following:
+
+- Discuss it with at least one other developer from a different squad – most appropriately through
+  the PR
+- Add as a talking point to the next FE sync so other developers have visibility
+
+You'd be surprised how often a library can be replaced with a native API and through thinking and
+acting in this way we hope to keep our dependency surface as small as possible.
+
+## Data fetching
+
+### V4
+
+Whilst we are actively developing V4 of the CMS we should _always_ be using `react-query`'s hooks to
+manage the fetching and mutation of data in conjunction with `useFetchClient`. This is because
+`react-query` caches our data-fetching meaning multiple calls are not necessarily made unless a
+mutation occurs as well as handling various intermittent states and events during the fetching
+lifecycle.
+
+:::note
+We're using `react-query` version 3, see the specific docs – https://tanstack.com/query/v3/docs/react/overview
+:::
+
```

Comment:
Could we add guidelines for the query cache key here?

- use the URL fragments from the called URL (`/admin/content-manger/:id/stages` -> `['content-manager', id, 'stages']` except `admin`
- ee calls should be prefixed with `ee`, to be able to invalidate it all at once one day
- all dynamic variables (including GET params) need to be part of the dependency array